deps/rabbitmq_mqtt/src/rabbit_mqtt.erl

```diff
@@ -69,7 +69,13 @@ emit_connection_info(Items, Ref, AggregatorPid, Pids) ->
     rabbit_control_misc:emitting_map_with_exit_handler(
       AggregatorPid, Ref,
       fun(Pid) ->
-              rabbit_mqtt_reader:info(Pid, Items)
+              case rabbit_mqtt_reader:info(Pid, Items) of
+                  {error, not_found} ->
+                      %% a Web-MQTT connection
```

Comment:
Are we sure there are no other ways to get `{error, not_found}`, or is it always a Web-MQTT connection?

Reply:
The logic seems to be "try obtaining connection metrics as if it was an MQTT connection, if not, do the same for Web MQTT". If both fail, then return `{error, not_found}`.

The original function returns `{error, not_found}` if so does `rabbit_mqtt_reader:info/2` so it seems tobe a safe thing to.

Reply:
if `Pid` is a `rabbit_mqtt_reader` gen_server (ie represents a regular MQTT connection), then there is no way it can return `{error, not_found}`. As Michael described then fall back to try Web-MQTT.

(if either of both calls exits with a "benign" reason, then `rabbit_control_misc:emitting_map_with_exit_handler` catches that and continues. Where "benign" is if the gen_server process does not exist, has shut down or cannot be contacted on a remote node)